### PR TITLE
Enable the Product Construction Service to connect with KeyVaults

### PR DESCRIPTION
**Pull request by dkurepa**
https://github.com/dotnet/arcade-services/issues/3163
Connects the Product Construction Service with it's environments Key Vault.
This allows us to use the secrets like `app.Configuration[<secret name>]`

In the staging environment, the service will authenticate using [Environmental Credentials](https://learn.microsoft.com/en-us/dotnet/api/azure.identity.environmentcredential?view=azure-dotnet). I created a new `app registration`, `product-construction-service-int`, gave it permissions to read secrets from ProductConstructionInt KV and put it's details in the Container App resource. The `DefaultAzureCredentails` will pick these up and connect to the vault. This is only a temporary solution tho, this approach isn't ideal because of two reasons:
 - The App secret will expire and will require manual rotation
 - The environment has to be set up by hand, as biceps doesn't offer any support for Entra ID apps

Once https://github.com/dotnet/arcade-services/issues/3180 is resolved, we'll be able to use Managed Identities for authentication. Managed Identities resolve both of the issues above, they require no secret rotation, and they can be fully configured in the bicep.

For local development, the `DefaultAzureCredentials` will pick up your VS credential, and use that to authenticate. Also we will have a different KV for this, ProductConstructionDev.

### Release Note Category
- [x] Feature changes/additions 
- [ ] Bug fixes
- [ ] Internal Infrastructure Improvements

### Release Note Description
Enable the Product Construction Service to connect with KeyVaults

### CODE REVIEW

**Comment by tkapin** [comment]
What is the proper solution if this is only a temporary one? Do we have the plan to get to the proper solution? Without this plan, we are introducing a technical debt from the beginning.

**Review by tkapin — DISMISSED** [review]
Comment left in the PR.

**Comment by dkurepa** [comment]
> What is the proper solution if this is only a temporary one? Do we have the plan to get to the proper solution? Without this plan, we are introducing a technical debt from the beginning.

I updated the PR description to better describe this. Basically, we're just waiting for the Container App team to enable a feature in WestUS2. When this happens, we will delete the current staging environment, and recreate it there, with our bicep script. The bicep script will take care of everything for us, including setting up the access for the Managed Identity that our Container App will use

File: src/ProductConstructionService/Readme.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Starting the service locally
 
-To run the Product Construction Service locally, set the `ProductConstructionService.AppHost` as Startup Project, and run with F5
+To run the Product Construction Service locally, set the `ProductConstructionService.AppHost` as Startup Project, and run with F5.
+
+When running locally, the service will attempt to read secrets from the `ProductConstructionDev` KeyVault, using your Microsoft credentials. If you're having some authentication double check the following:
```
Comment (premun) [inline src/ProductConstructionService/Readme.md:5]: Can you make the KV name a link to Azure?

**Review by premun — APPROVED** [review]

**Comment by dkurepa** [comment]
> @dkurepa how is it set that we will be able to access the Dev KV using our identity? Is there some group representing our team added to the permissions? Or it's inherited from `HelixStaging` permissions?

It's inherited from `HelixStaging` we don't have to do anything. At least not while we're in `HelixStaging`

File: src/ProductConstructionService/Readme.md (hunk 1 above)
Comment (dkurepa) [reply src/ProductConstructionService/Readme.md]: Added, just need to make sure to update the url when we move to westus

**Review by premun — APPROVED** [review]

**Pull request merged**